File: c++/kattis/planktonfarm/planktonfarm.cpp
#include <bits/stdc++.h>

using namespace std;

int main () {
  long a, b, c[1000000];
  cin >> a >> b;
  for (int i = 0; i < b; i++) {
    long curr;
    cin >> curr;
    c[i] = curr;
  }
  sort(c, c + b);
  long p1=0, p2=b-1, ans=0;
  while (p1 < p2) {
    if (c[p1] * c[p2] < a) {
      p1++;
    } else {
      p1++;
      p2--;
      ans++;
    }
  }
  cout << ans << endl;
}

File: java/kattis/planktonfarm/Planktonfarm.java
import java.util.*;
import java.io.*;

class Planktonfarm {
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        int[] c = new int[b];
        
        for (int i = 0; i < b; i++) {
            int curr = sc.nextInt();
            c[i] = curr;
        }
        
        Arrays.sort(c);

        int p1=0, p2=b-1, ans=0;
        while (p1 < p2) {
            if (c[p1] * c[p2] >= a) {
                p2--;
                ans++;
            }
            p1++;
        }
        System.out.println(ans);
    }
}
